components/tor/BUILD.gn

```diff
@@ -0,0 +1,93 @@
+import("//brave/components/tor/buildflags/buildflags.gni")
+
+source_set("tor") {
+  public_deps = [
+    "//brave/components/tor/buildflags",
+  ]
+
+  sources = [
+    "tor_constants.cc",
+    "tor_constants.h",
+    "tor_launcher_service_observer.h",
+  ]
+
+  deps = [
+    ":pref_names",
+    ":switches",
+    "//base",
+  ]
+
+  if (enable_tor) {
+    sources += [
+      "brave_tor_client_updater.cc",
+      "brave_tor_client_updater.h",
+      "service_sandbox_type.h",
+      "tor_control.cc",
+      "tor_control.h",
+      "tor_control_event.cc",
+      "tor_control_event.h",
+      "tor_control_event_list.h",
+      "tor_launcher_factory.cc",
+      "tor_launcher_factory.h",
+      "tor_navigation_throttle.cc",
+      "tor_navigation_throttle.h",
+      "tor_profile_service.cc",
+      "tor_profile_service.h",
+      "tor_profile_service_impl.cc",
+      "tor_profile_service_impl.h",
+      "tor_tab_helper.cc",
+      "tor_tab_helper.h",
+      "onion_location_navigation_throttle.cc",
+      "onion_location_navigation_throttle.h",
+      "onion_location_tab_helper.cc",
+      "onion_location_tab_helper.h",
+    ]
+  }
+  
+  deps += [
+    "//brave/app:brave_generated_resources_grit",
```

Comment:
I think this is layer violoation. If tor components needs resources, it should be included in component resource.

Reply:
addressed in e1726d757df0524a90d1c7a961847bebc7ed8eee